stl/src/syserror_import_lib.cpp

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+// This must be as small as possible, because its contents are
+// injected into the msvcprt.lib and msvcprtd.lib import libraries.
+// Do not include or define anything else here.
+// In particular, basic_string must not be included here.
+
+#include <__msvc_system_error_abi.hpp>
+
+#include <Windows.h>
+
+namespace {
+
+    struct _Whitespace_bitmap_t {
+        bool _Is_whitespace[256];
+
+        constexpr _Whitespace_bitmap_t() : _Is_whitespace{} {
+            _Is_whitespace[' ']  = true;
+            _Is_whitespace['\n'] = true;
+            _Is_whitespace['\r'] = true;
+            _Is_whitespace['\t'] = true;
+            _Is_whitespace['\0'] = true;
+        }
+
+        constexpr bool _Test(const char _Ch) const noexcept {
+            return _Is_whitespace[static_cast<unsigned char>(_Ch)];
+        }
+    };
+
+    constexpr _Whitespace_bitmap_t _Whitespace_bitmap;
+
+} // unnamed namespace
+
+_EXTERN_C
+
+_NODISCARD size_t __CLRCALL_PURE_OR_STDCALL __std_get_string_size_without_trailing_whitespace(
+    _In_ const char* const _Str, _In_ size_t _Size) noexcept {
+    for (;;) {
+        if (_Size == 0) {
+            return _Size;
+        }
+
+        --_Size;
+        if (!_Whitespace_bitmap._Test(_Str[_Size])) {
+            ++_Size;
+            return _Size;
+        }
+    }
+}
+
+_NODISCARD size_t __CLRCALL_PURE_OR_STDCALL __std_system_error_allocate_message(
+    _In_ const unsigned long _Message_id, _Out_ char** const _Ptr_str) noexcept {
+    // convert to name of Windows error, return 0 for failure, otherwise return number of chars in buffer
+    // __std_system_error_free_message should be called even if 0 is returned
+    // pre: *_Ptr_str == nullptr
+    const unsigned long _Chars =
+        FormatMessageA(FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS,

```

Comment:
[FormatMessageA documentation](https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-formatmessagea) says for `FORMAT_MESSAGE_ALLOCATE_BUFFER`: "In previous versions of Windows, this value was not available for use when compiling Windows Store apps. As of Windows 10 this value can be used."

Is this an issue?

Reply:
I asked Win32PRG.

Reply:
Windows 8.1 Store app support was dropped in VS 2017, so I don't think it'll be an issue for a standard library shipping in VS 2019.

Reply:
@sylveon The IDE doesn't support building new apps for that OS, but the same redist / support DLLs are used all the way back to 2015 (since we didn't break ABI from that release). Note our support page where we call it the "2015-2019 redist": https://support.microsoft.com/en-us/help/2977003/the-latest-supported-visual-c-downloads

@StephanTLavavej I asked on WIN32PRG internally and got this reply from Raymond Chen (yes that one):

> This is a WACK thing. The text should be more like
>
> * During the Windows 8 and 8.1 timeframe, the Store did not accept apps that used the `FORMAT_MESSAGE_ALLOCATE_BUFFER` flag. Coincident with the release of Windows 10, the Store began accepting apps that use this flag.
>
> It’s purely a Store policy thing, not an OS functionality thing.

so I think we're good.